src/main/java/com/mongodb/hadoop/output/MongoRecordWriter.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2011-2013 10gen Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hadoop.output;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBCollection;
+import com.mongodb.DBObject;
+import com.mongodb.hadoop.MongoOutput;
+import com.mongodb.hadoop.io.BSONWritable;
+import com.mongodb.hadoop.io.MongoUpdateWritable;
+import com.mongodb.hadoop.io.MongoWritableTypes;
+import com.mongodb.hadoop.util.MongoConfigUtil;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.bson.BSONObject;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+  * Create a MongoRecordWriter targeting a single DBCollection.
+  * @param <K> the type of the key
+  * @param <V> the type of the value
+*/
+public class MongoRecordWriter<K, V> extends RecordWriter<K, V> {
+
+    private static final Log LOG = LogFactory.getLog(MongoRecordWriter.class);
+    private final DBCollection collection;
+    private final TaskAttemptContext context;
+    private final BSONWritable bsonWritable;
+    private FSDataOutputStream outputStream;
+
+    public MongoRecordWriter(
+      final DBCollection c,
+      final TaskAttemptContext ctx) {
```

Comment:
All uses of `compatUtil` have been removed (e.g., https://github.com/mongodb/mongo-hadoop/blob/20208a027ad8638e56dfcf040773f176d6ee059f/core/src/main/java/com/mongodb/hadoop/output/MongoRecordWriter.java#L51)